File: Java/Day08_Numberpyramid.java

/**
 * Day 08: Number Pyramid
 * Printing a pyramid with numbers where each row contains the row number (1, 2 2, 3 3 3...).
 */

import java.util.*;

public class Day08_Numberpyramid {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the value of n:");
        int n = sc.nextInt();

        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n - i; j++) {
                System.out.print(" ");
            }
            for (int j = 1; j <= i; j++) {
                System.out.print(i + " ");
            }
            System.out.println();
        }
    }
}

File: Python/Day08_Numberpyramid.py
"""
Day 08: Number Pyramid
Printing a pyramid with numbers where each row contains the row number (1, 2 2, 3 3 3...).
"""

n = int(input("Enter the value of n:"))

for i in range(1, n + 1):
    # Spaces
    for j in range(1, n - i + 1):
        print(" ", end="")
        
    # Numbers (print i repeatedly) followed by space
    for j in range(1, i + 1):
        print(str(i) + " ", end="")
    print()
